src/presentation/http/router/noteSettings.ts

```diff
@@ -89,6 +90,45 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(noteSettings);
   });
 
+  /**
+   * patch team member role by user and note id
+   *
+   * TODO add policy for this route to check id user have 'write' role if this team to patch someone's else role
+   */
+  fastify.patch<{
+    Params: {
+      notePublicId: NotePublicId,
+      userId: User['id'];
+      newRole: keyof typeof MemberRole,
```

Comment:
Maybe we should make separate type for `keyof typeof MemberRole`?

Reply:
we can use just "write" | "read" , or make new type?

Reply:
done